### PR TITLE
feat(producer): allow disabling pendingMessagesQueue

### PR DESCRIPTION
**Pull request by Eywek**

### CODE REVIEW

**Review by galrose — CHANGES_REQUESTED** [review]
LGTM, could you just add a test for the happy flow of writing with the queue size on zero?
Just to make sure its not broken in the future

**Comment by Eywek** [comment]
> LGTM, could you just add a test for the happy flow of writing with the queue size on zero? Just to make sure its not broken in the future

done!

File: test/producer/index-spec.js [hunk 1]
```diff
@@ -281,6 +281,38 @@ describe('Producer tests', function () {
       await producer.close();
     });
   });
+  describe('on connection exception should not resend batch', function () {
+    it('should not throw exception', async function () {
```
Comment (galrose) [inline test/producer/index-spec.js:285]: The test you created checks that you catch the error it doesn't check that data was sent using the producer, some of the tests above test actual sending. For example, the it 'should resend messages after unload' test is a simple test that checks sending messages. (You don't need to unload just check that sending messages works in  general.)
Can you rename the describe and the it to something more intuitive to what the test does and specifically about the queue size being 0?

**Review by galrose — CHANGES_REQUESTED** [review]
